modules/storages/app/models/storages/storage.rb

```diff
@@ -104,4 +105,8 @@ def short_provider_type
   def provider_type_nextcloud?
     provider_type == ::Storages::Storage::PROVIDER_TYPE_NEXTCLOUD
   end
+
+  def provider_type_one_drive?
+    provider_type == ::Storages::Storage::PROVIDER_TYPE_ONE_DRIVE
+  end
 end
```

Comment:
🟡 WDYT of making this into a query method that takes in a parameter? Also this is basically a type check. Can't we substitute its uses by duck typing?

```ruby
def provider_type?(storage_klass)
  self.class == storage_klass
end
```

Reply:
What is the advantage of this method over simply using `==`? The advantage of having the explicit check is less characters in the calling places:

```ruby
<% if @storage.provider_type_nextcloud? %>

<% if @storage == ::Storages::Storage::PROVIDER_TYPE_NEXTCLOUD %>
```

so, this method is just for convenience for callers.

Reply:
Yes, but this can blow up once we have multiple provider types. There must be a way to not have to add this on a per provider basis. Right now we need to add it to a list of constants and add a method. =/

Reply:
right, but what is the alternative? we can remove them and use `==` everywhere. No big reason to not do it. The provider types are in the end those longer variables. These bool methods are in the end not needed, they are just sugar.

Can you think of a convenient form, that is not using the original variable names? what about dynamic method generation: we iterate over the provider types array and generate a method `provider_type_{short_name}?` for every type.

Reply:
That's not a hill I'm willing to die on. :rofl: I'll see if I can come up with an idea later on. :P